20180321/README.md

```diff
@@ -1 +1,188 @@
+
+  pod 'AlpsSDK', '~> 0.6'
+  ...
+```
+
+After that execute `pod install`, this will download AlpsSDK and generate `my-app.xcworkspace` that you have to import in Xcode (the old my-app.xcodeproj will not work correctly).
+
+After importing `my-app.xcworkspace` into Xcode, we will add necessary permissions for the app in `info.plist` file. Right click on `info.plist` select `Open As > Source Code` and add two keys at the end:
+
+```
+...
+    <key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
+    <string>This app needs your current location</string>
+    
+    <key>NSLocationWhenInUseUsageDescription</key>
+    <string>This app needs your current location</string>
+</dict>
+</plist>
+```
+
+## SDK configuration
+Next we have to initialize SDK. To do that in `AppDelegate.swift` in `application` funtion add thos two lines:
+```swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+
+  let config = MatchMoreConfig(apiKey: "paste_API_key_from_portal_here")
+  MatchMore.configure(config)
+
+  return true
+}
+```
+The first line creates a configuration with your API key second configures AlpsSDK.
+There will be compilation error because Xcode does not know what `MatchMoreConfig` is, to fix that add `import AlpsSDK` import.
+
+Now that SDK is set up we will create a label and two buttons in main application view in `Main.storyboard`.
+
+![alt text](https://raw.githubusercontent.com/matchmore/tech-blog/master/20180321/img/app-main-page.png "app main view")
+
+## Create publication
+Let's write the code that will create publication and subscription when clicking on buttons.
+
+So first you create `IBAction` for publish button:
+```swift
+@IBAction func publish() {
+    MatchMore.createPublicationForMainDevice(publication: Publication(topic: "Test Topic", range: 20, duration: 100, properties: ["test": "true"]), completion: { result in
+        switch result {
+        case .success(let publication):
+            print("🏔 Pub was created: 🏔\n\(publication.encodeToJSON())")
+        case .failure(let error):
+            print("🌋 \(String(describing: error?.message)) 🌋")
+        }
+    })
+}
+```
+
+This function will create publication attached to phone location with range 20 meters and duration 100 seconds (will be removed after 100 seconds). This publication will be on topic `Test Topic` which means that it will only match with subscriptions on the same topic. It also has a property named `test` set to `true`, publications can have an arbitrary number of properties with arbitrary names that can be used to store information in them and to restrict further when the match should happen by adjusting selector in the subscription.
+
+## Create subscription
+Next is `IBAction` for subscribe button:
+```swift
+@IBAction func subscribe() {
+    let subscription = Subscription(topic: "Test Topic", range: 20, duration: 100, selector: "test = true")
```

Comment:
interesting that works. It should be `"test = 'true'"` if properties above like as described.

If you change properties as suggested than it's fine.

Reply:
now it is boolean